evm_arithmetization/src/witness/operation.rs

```diff
@@ -397,7 +448,16 @@ pub(crate) fn generate_set_context<F: Field>(
         );
         (sp_to_save, op)
     } else {
-        mem_read_with_log(GeneralPurpose(2), new_sp_addr, state)
+        // Even though we might be in the interpreter, `Stack` is not part of the
+        // preinitialized segments, so we don't need to carry out the additional checks
+        // when get the value from memory.
```

Comment:
These comments are just to explain why `is_interpreter` is set to false even though we may be in the interpreter. So I think they are necessary, as otherwise, the `false` in the arguments might be confusing